### ISSUE TITLE
Duplicate Flow Runs Created at the End of Daylight Savings Time

### ISSUE THREAD

**Issue by biancaines** [PrefectHQ/prefect#7461]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

Two users reported that their flow runs were running in duplicate following the end of daylight savings time.
For the first user, they provided the following details:

> Rich Tata
 [Yesterday at 4:42 PM](https://prefect-community.slack.com/archives/CL09KU1K7/p1667770945860739)
So DST caused some havoc in my flows today. After the time change, every flow has been running two times, once at it's normal time, and again one hour in the future. I set them using cron string in my yaml and adjusting the timezone to my local and thought based on documentation that it would auto adjust (pictured).

> Pausing and unpausing all deployments fixed my deployment schedules for all future runs. Still, did anyone else encounter this issue? Was this my error or a bug?

Examples of their affected flow run IDs are as follows:

First set of duplicates:
`78b27915-bc51-4b59-bc9c-8ff405c93973`
`0edb8d60-fd59-433c-a995-895c4e0c3f23`

Second set of duplicates:
`19eabb57-d58f-4c4e-b5af-8c790ca74f44`
`85c4a62b-1021-4c96-995c-98d0c4664468`

Third set of duplicates:
`5eccbbc7-dc08-4ab1-ade2-02c71bd1a203`
`b60d73cb-2f10-4cab-91a8-63078a0b58b7`

For the second user, they stated the following:
> Aaron
  [1 hour ago](https://prefect-community.slack.com/archives/CL09KU1K7/p1667835035378569?thread_ts=1667770945.860739&cid=CL09KU1K7)
I had the exact same issue, jobs ran twice an hour apart. Mine were deployed at the command line with Python, and then scheduled in the UI with cron
![image](https://user-images.githubusercontent.com/105657837/200361354-09603904-7e87-458c-8366-5f04c96dd85c.png)

Examples of their affected flow run IDs are as follows:
`e87d232b-990f-45a0-b4bf-71a21d5c2e8b ran at 8am` 
`867b40a4-824a-438d-b546-c5af067cc60e ran at 7am`

### Reproduction

"""The first user provided the following information to reproduce the error:
We use CLI to create and apply deployments once the code is ready.
1. prefect deployment build ./<FILENAME>.py:<FLOWFUNCTIONNAME> -n <DEPLOYMENTNAME> -q <WORKQUEUE> --cron "* * * * *" -t "<TAG1>" -o "../yaml/<CHANGETHIS>-deployment.yaml"

a. We use this command template to generate a yaml; cron is specified there.
b. the Timezone is set to America/Detroit in the cron file.

2. prefect deployment apply ../yaml/<CHANGETHIS>-deployment.yaml

a. Once we're sure it looks right, we use this to deploy to prefect.
b. verify everything is right (workqueue, schedule, tags) in the prefect UI.

The second user reported that they deployed their flows at using the command line with Python, and applied the schedule through the UI using cron (see Bug summary section for screenshot)."""

### Error

```python3
Flow runs are running in duplicate following the end of daylight savings time.
```

### Versions

```Text
First user: 
-Perfect version: 2.3.2
-API version: 0.8.0
-Python version: 3.9.13

Second user:
-Prefect version: 2.2.0
```

### Additional context

Appears to be related to the following issue:
#7381

**Comment by discdiver** [comment]
Looks like a duplicate of #7381

**Comment by madkinsz** [comment]
@discdiver there's a separate "close" in the dropdown for when it's a duplicate.

**Comment by discdiver** [comment]
Ah,my bad.  Thanks.
